presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/thrift/MockThriftMetastoreClient.java

```diff
@@ -218,7 +218,8 @@ public void deleteTableColumnStatistics(String databaseName, String tableName, S
     @Override
     public void setPartitionColumnStatistics(String databaseName, String tableName, String partitionName, List<ColumnStatisticsObj> statistics)
     {
-        throw new UnsupportedOperationException();
+        accessCount.incrementAndGet();
+        // No-op
```

Comment:
the methods here are either not implement (throw) or implemented.

Reply:
@sopel39 @electrum what about replacing `MockThriftMetastoreClient` with `InMemoryThriftMetastore` (+ access counting wrapper) in `TestCachingHiveMetastore`?

Reply:
Yes, I thought of doing that as well - but chose this instead since the alter/set-stats operation was not being used anywhere as of now. there are some no-ops around roles as well in `MockThriftMetastoreClient`

Reply:
> @sopel39 @electrum what about replacing MockThriftMetastoreClient with InMemoryThriftMetastore (+ access counting wrapper) in TestCachingHiveMetastore?

It seems like a good idea. I don't think `MockThriftMetastoreClient` correctly mocks thrift metastore. I would say this could be a follow up PR though.